lib/src/query.dart

```diff
@@ -351,7 +351,7 @@ class PostgreSQLFormatIdentifier {
 
       final dataTypeString = variableComponents.last;
       try {
-        type = typeStringToCodeMap[dataTypeString]!;
+        type = typeStringToCodeMap[dataTypeString];
```

Comment:
The test that is failing is due to this line of code.

Apparently, the code was throwing an error whenever `type` is `null` by relying on the removed null check -- that is, when the null check fails, it throws a format exception. 

I think a proper way to handle this is to replace the try/catch statement with if/throw such as:

```dart
      type = typeStringToCodeMap[dataTypeString];
      if (type == null) {
        throw FormatException(
            "Invalid type code in substitution variable '$t'");
      }
```

Reply:
This looks good to me. @eseidel could you please rebase the PR and also fix this code?